src/cli/mod.rs

```diff
@@ -0,0 +1,85 @@
+//! Command-line interface (CLI) logic.
+
+use crate::ipc::ClientToDaemonConnection;
+use anyhow::{Context, Result};
+use structopt::StructOpt;
+
+/// Top-level CLI options.
+#[derive(Debug, StructOpt)]
+pub struct CliOptions {
+    /// CLI sub-commands.
+    #[structopt(subcommand)]
+    pub(crate) cmd: CliCommand,
+}
+
+impl CliOptions {
+    /// Run CLI application.
+    pub fn run(self) -> Result<()> {
+        match self.cmd {
+            CliCommand::Backend(opts) => run_backend(opts),
+            CliCommand::Daemon => crate::daemon(),
+            CliCommand::Status(opts) => run_status(opts),
+            CliCommand::Update => run_update(),
+        }
+    }
+}
+
+/// CLI sub-commands.
+#[derive(Debug, StructOpt)]
+pub(crate) enum CliCommand {
+    #[structopt(name = "backend")]
+    Backend(crate::BackendOpt),
+    #[structopt(name = "daemon")]
+    Daemon,
```

Comment:
Part of the intent before was that these options were hidden.  That said perhaps what we really want to do is go the hardlink path again with `/usr/libexec/bootupd` and dispatch again based on `argv[0]`.

Reply:
The multiexec approach looks cleaner. How do you want to bucket the sub-commands?
 1. `bootupd daemon` ?
 1. `bootupctl status|update` ?
 1. where does `backend` belong?